compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/GenerateClosuresForDefaultValues.java

```diff
@@ -187,29 +189,31 @@
 import static org.ballerinalang.model.symbols.SymbolOrigin.VIRTUAL;
 
 /**
- * Parameter desugar for create closures for default values.
+ * GenerateClosuresForDefaultValues for create closures for default values.
  *
  * @since 2201.3.0
  */
-public class ParameterDesugar extends BLangNodeVisitor {
-    private static final CompilerContext.Key<ParameterDesugar> PARAMETER_DESUGAR_KEY = new CompilerContext.Key<>();
+public class GenerateClosuresForDefaultValues extends BLangNodeVisitor {
```

Comment:
Can't we implement this as a `org.wso2.ballerinalang.compiler.tree.SimpleBLangNodeAnalyzer` instead? Can pass temporary fields via data. We can do this separately though.

Reply:
Please create an issue and mention the issue here.

Reply:
https://github.com/ballerina-platform/ballerina-lang/issues/40884